src/Symfony/Component/ObjectMapper/Attribute/Map.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Attribute;
+
+/**
+ * Configures a class or a property to map to.
+ *
+ * @experimental
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ *
+ * @psalm-type CallableType =
+ */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+readonly class Map
+{
+    /**
+     * @param string|class-string|null                                      $source    The property or the class to map from
+     * @param string|class-string|null                                      $target    The property or the class to map to
+     * @param string|bool|callable(mixed $value, object $object): bool|null $if        A boolean, Symfony service name or a callable that instructs whether to map
+     * @param (string|callable(mixed $value, object $object): mixed)|(string|callable(mixed $value, object $object): mixed)[]|null                              $transform A Symfony service name or a callable that transform the value during mapping
```

Comment:
```suggestion
     * @param CallableType|CallableType[]|null                              $transform A Symfony service name or a callable that transform the value during mapping
```

Reply:
using a `@param` annotation with a psalm type alias is a no-go. Any tool not supporting psalm aliases will consider `CallableType` as a class name, which would totally break the documented type of our public API.
Type aliases require using a prefixed annotation (if a tool is reading a `@psalm-param` annotation for its own usage, it is expected to support all the psalm-specific types).

I would actually suggest to keep the existing `@param` and to remove the `@psalm-type` entirely.